Muvr/Entities/MRManagedClassifiedExercise+CoreDataAnalytics.swift

```diff
@@ -0,0 +1,172 @@
+                averageIntensity: averageIntensity / const,
+                averageRepetitions: Int(Double(averageRepetitions) / const),
+                averageWeight: averageWeight / const,
+                averageDuration: averageDuration / const
+            )
+        }
+    }
+    
+    ///
+    /// Expression description for ``COUNT(_objectId)``, where ``_objectId`` is the identifier
+    /// of whatever entity it is being applied to.
+    ///
+    private static let countByEntity: NSExpressionDescription = {
+        let count = NSExpressionDescription()
+        count.name = "count"
+        count.expression = NSExpression(forFunction: "count:", arguments: [NSExpression(forKeyPath: "entity")])
+        count.expressionResultType = NSAttributeType.Integer32AttributeType
+        return count
+    }()
+    
+    ///
+    /// Expression description for ``average:`` of ``keyPath``.
+    /// - parameter keyPath: the property's KP
+    /// - returns: the expression description
+    ///
+    private static func averageFor(keyPath: String) -> NSExpressionDescription {
+        let expr = NSExpressionDescription()
+        expr.name = "\(keyPath)"
+        expr.expression = NSExpression(forFunction: "average:", arguments: [NSExpression(forKeyPath: keyPath)])
+        expr.expressionResultType = NSAttributeType.DoubleAttributeType
+        return expr
+    }
+    
+    ///
+    /// TODO: this is a bit odd. It would be best to define MKExerciseId a full
+    /// TODO: type, not just typealias, and then add this to the new type.
+    ///
+    /// Returns a component of the exerciseId at the given index
+    /// - parameter index: the index 0..(number of slashes)
+    /// - parameter exerciseId: the exercise id
+    /// - returns: the element at the given index
+    ///
+    private static func componentAt(index: Int, exerciseId: String) -> String? {
+        let components = exerciseId.componentsSeparatedByString("/")
+        if index >= 0 && index < components.count { return components[index] }
+        return nil
+    }
+
+    ///
+    /// Computes the averages for the ``MRManagedClassifiedExercise`` in the given ``managedObjectContext``,
+    /// whose ``exerciseId`` starts with ``exerciseIdPrefix``, dropping the ``exerciseIdPrefix`` from the
+    /// exerciseIds returned.
+    /// - parameter managedObjectContext: the MOC
+    /// - parameter exerciseIdPrefix: the exercise id prefix to match (i.e. back/, arms/)
+    /// - returns: the average for the last 100 sessions
+    ///
+    static func averages(inManagedObjectContext managedObjectContext: NSManagedObjectContext, exerciseIdPrefix: String?) -> [Average] {
+        let fetchLimit = 100
+        let entity = NSEntityDescription.entityForName("MRManagedClassifiedExercise", inManagedObjectContext: managedObjectContext)
+        let exerciseId = entity!.propertiesByName["exerciseId"]!
+
+        let fetchRequest = NSFetchRequest(entityName: "MRManagedClassifiedExercise")
```

Comment:
Why not loading ``MRManagedLabelledExercise`` as well ?

May be hard to combine but needed to be able to use the phone without a watch.